src/server/tran_server.hpp

```diff
@@ -96,11 +96,6 @@ class tran_server
     virtual bool get_remote_storage_config () = 0;
     virtual void on_boot () = 0;
 
-    // Common request Handlers
-    void receive_boot_info (cubpacking::unpacker &upk);
-    void receive_log_page (cubpacking::unpacker &upk);
-    void receive_data_page (cubpacking::unpacker &upk);
-
     virtual request_handlers_map_t get_request_handlers () = 0;
```

Comment:
Here we have an example of a function that is marked pure but also has an implementation. :)

Reply:
missed this

Reply:
@adrianbrab
There was no problem with the code. It was valid. I merely pointed out a rarely occurring situation. It's perfectly valid and even needed in situation where it is needed to have an implementation but also a need to mark the class as non-instantiable. It's the downside of c++ not having interfaces.